# INFERENTZIA ESTATISTIKOA: Bigarren praktika

## Estimazioak eta konfiantza-tarteak (KTak)


In [128]:
# txt datuak irakurri

Datuak <-read.table(file="altuerak.txt", header=TRUE, dec=",", sep="\t")
Datuak

AITARENA AMARENA SEXUA       JAIOTERRIA ALTUERA PISUA
1   174      156     emakumezkoa 2          165      65  
2   177      159     gizonezkoa  2          170      67  
3   173      161     gizonezkoa  1          168      51  
4   174      156     gizonezkoa  2          167      69  
5   160      165     gizonezkoa  1          162      54  
6   167      157     gizonezkoa  3          163      61  
7   171      162     emakumezkoa 1          165      65  
8   174      158     gizonezkoa  3          168      76  
9   175      162     emakumezkoa 3          168      67  
10  174      161     emakumezkoa 1          167      69  
11  175      162     emakumezkoa 3          168      66  
12  176      161     gizonezkoa  2          170      81  
13  180      156     gizonezkoa  2          170      92  
14  172      171     emakumezkoa 3          170      85  
15  171      158     emakumezkoa 1          164      58  
16  172      160     emakumezkoa 3          165      65  
17  172      157     gizonezkoa  2          166      72  
18  175      158     emakumezkoa 2          166      62  
19  173      160     gizonezkoa  2          168      86  
20  173      164     emakumezkoa 1          165      65  
21  171      167     emakumezkoa 2          167      69  
22  177      160     emakumezkoa 2          168      74  
23  172      161     emakumezkoa 3          166      62  
24  180      164     gizonezkoa  1          174     108  
25  174      159     gizonezkoa  3          168      75  
26  169      167     gizonezkoa  2          168      77  
27  174      166     emakumezkoa 2          169      63  
28  166      160     gizonezkoa  2          164      68  
29  168      159     emakumezkoa 3          162      44  
30  172      163     emakumezkoa 2          166      72  
⋮   ⋮        ⋮       ⋮           ⋮          ⋮       ⋮    
142 174      159     gizonezkoa  2          168     66   
143 177      161     gizonezkoa  3          171     87   
144 180      160     emakumezkoa 1          170     80   
145 168      163     gizonezkoa  3          166     77   
146 168      164     emakumezkoa 3          164     58   
147 171      160     gizonezkoa  1          167     69   
148 170      162     emakumezkoa 3          165     65   
149 157      162     gizonezkoa  3          159     33   
150 163      162     gizonezkoa  2          163     61   
151 166      165     emakumezkoa 2          164     58   
152 166      163     emakumezkoa 2          163     51   
153 169      166     gizonezkoa  2          168     69   
154 175      158     gizonezkoa  1          168     67   
155 172      157     emakumezkoa 2          164     58   
156 170      161     gizonezkoa  3          166     82   
157 177      158     emakumezkoa 2          167     79   
158 175      155     gizonezkoa  2          167     79   
159 174      158     gizonezkoa  2          168     68   
160 173      166     emakumezkoa 3          168     86   
161 176      157     emakumezkoa 2          166     72   
162 162      161     gizonezkoa  2          162     54   
163 174      158     gizonezkoa  2          168     66   
164 169      161     emakumezkoa 3          164     58   
165 174      156     emakumezkoa 1          165     65   
166 165      170     emakumezkoa 2          165     65   
167 169      164     gizonezkoa  3          167     79   
168 166      157     emakumezkoa 3          160     30   
169 173      159     emakumezkoa 2          165     65   
170 176      158     gizonezkoa  1          169     82   
171 180      161     gizonezkoa  3          172     84

### Populazioaren pisuaren batezbestekoa: estimazioa eta %95eko KTa

In [129]:
t.test(Datuak$PISUA,alternative='two.sided',conf.level=.95)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 mean = 0

Alternative Hypothesis:          True mean is not equal to 0

Test Name:                       One Sample t-test

Estimated Parameter(s):          mean of x = 72.54386

Data:                            Datuak$PISUA

Test Statistic:                  t = 52.95709

Test Statistic Parameter:        df = 170

P-value:                         1.403626e-107

95% Confidence Interval:         LCL = 69.83973
                                 UCL = 75.24799


#### Azalpena:
Populazioaren bateztebesteko pisuaren estimatutako balioa 72.54386 da eta %95eko KTa (69.83973,75.24799) da

#### Oharra: Beste konfiantza-mailak egiteko
Adibidez %99ko KTa egiteko, conf.level=.95 jarri ordez, conf.level=.99

### Populazioaren altueraren bariantza/desbideratze estandarra: estimazioa eta %95eko KTa

In [130]:
# agian hurrengo paketea instalatu eta kargatu behar duzue 
#install.packages("EnvStats")
#library(EnvStats)
varTest(Datuak$ALTUERA,alternative  = "two.sided",conf.level=0.95)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 variance = 1

Alternative Hypothesis:          True variance is not equal to 1

Test Name:                       Chi-Squared Test on Variance

Estimated Parameter(s):          variance = 11.08242

Data:                            Datuak$ALTUERA

Test Statistic:                  Chi-Squared = 1884.012

Test Statistic Parameter:        df = 170

P-value:                         0

95% Confidence Interval:         LCL =  9.057947
                                 UCL = 13.874455


#### Azalpena:
Populazioaren altueraren bariantzarako estimatutako balioa 11.08242 da eta %95eko KTa (9.353491, 13.376088) da

#### Oharra: Desbideratze estandarraren konfiantza-mailak egiteko
Bariantzaren ordez, desbideratze estandarraren balioak lortzeko sqrt funtzioa erabili ahal da.

In [131]:
# Populazioaren altueraren desbideratze estandarraren estimatutako balioa 
sqrt(11.08242)
# KTaren behe muga 
sqrt(9.353491)
# KTaren goi mugu
sqrt(13.376088)

[1] 3.329027

[1] 3.058348

[1] 3.657333

## Bi populazio askeren bariantzen zatidura

### Gizonezkoen eta emakumezkoen altueren bariantzen zatiduraren KTa

In [132]:
emakumezkoa <-Datuak$ALTUERA[Datuak$SEXUA=="emakumezkoa"]#c(103,94,110,87,98)       
gizonezkoa  <-Datuak$ALTUERA[Datuak$SEXUA=="gizonezkoa"]#c(97, 82,123,92,175,88,118)
var.test(emakumezkoa,gizonezkoa,conf.level=0.9)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 ratio of variances = 1

Alternative Hypothesis:          True ratio of variances is not equal to 1

Test Name:                       F test to compare two variances

Estimated Parameter(s):          ratio of variances = 0.8843092

Data:                            emakumezkoa and gizonezkoa

Test Statistic:                  F = 0.8843092

Test Statistic Parameters:       num df   = 81
                                 denom df = 88

P-value:                         0.5759727

90% Confidence Interval:         LCL = 0.6177924
                                 UCL = 1.2697464


#### Azalpena:
Populazioaren emakumeen ta gizonezkoen altueren bariantzen zatidurarako estimatutako balioa 0.8843092 da eta %95eko KTa (0.6177924, 1.2697464) da. Hortaz, ezin dugu ondorioztatu emakumezkoen eta gizonezkoen altueren bariantzak ezberdinak direnik.

#### Oharra: Desbideratze estandarraren konfiantza-mailak egiteko
Bariantzaren ordez, desbideratze estandarraren balioak lortzeko sqrt funtzioa erabili ahal da.

## Bi populazio askeren batezbestekoen diferentzia

### Gizonezkoen eta emakumezkoen altueren batezbestekoaren diferentziaren KTa

In [133]:
# aurreko atalean ikusi dugu bariantzak ezberdinak ez direla ezin dugula ondorioztatu. 
# Horregatik, var.equal=FALSE jartzen dugu. Aurreko ataleko ondorioa kontrakoa balitz
#   (bariantzak ezberdinak izatea), orduan var.equal=TRUE
t.test(emakumezkoa,gizonezkoa,conf.level=0.9,var.equal=FALSE)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 difference in means = 0

Alternative Hypothesis:          True difference in means is not equal to 0

Test Name:                       Welch Two Sample t-test

Estimated Parameter(s):          mean of x = 165.7805
                                 mean of y = 167.7640

Data:                            emakumezkoa and gizonezkoa

Test Statistic:                  t = -4.0768

Test Statistic Parameter:        df = 168.926

P-value:                         7.01739e-05

90% Confidence Interval:         LCL = -2.788270
                                 UCL = -1.178844


#### Azalpena:
Emakumezkoen batezbesteko altueraren estimazioa: 165.7805

Gizonezkoen batezbesteko altueraren estimazioa: 165.7805

Kenketaren KTa: (-2.788270,-1.178844)

#### Ondorioa:
Lortutako KTaren balioak negatiboak dira, orduan %90eko konfiantza-mailaz mu1-mu2 (hau da, 
emakumezkoen batezbesteko altuera ken gizonezkoen bateztebesteko altuera)
negatiboa dela baieztatzen dugu eta, orduan, emakumezkoen batezbesteko altuera gizonezkoen bateztebesteko altuera baino txikiagoa dela ondorioztatzen dugu.

## Binakako datuen KTa

### Aitaren eta amaren altueren arteko diferentzia

In [134]:
t.test(Datuak$AITARENA,Datuak$AMARENA, conf.level=0.98, var.equal=FALSE)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 difference in means = 0

Alternative Hypothesis:          True difference in means is not equal to 0

Test Name:                       Welch Two Sample t-test

Estimated Parameter(s):          mean of x = 172.1871
                                 mean of y = 160.8830

Data:                            Datuak$AITARENA and Datuak$AMARENA

Test Statistic:                  t = 23.91367

Test Statistic Parameter:        df = 316.3426

P-value:                         6.750131e-73

98% Confidence Interval:         LCL = 10.19882
                                 UCL = 12.40937


#### Azalpena:
Emakumezkoen batezbesteko altueraren estimazioa: 172.1871

Gizonezkoen batezbesteko altueraren estimazioa: 160.8830

Kenketaren KTa: (10.19882,12.40937)

#### Ondorioa:
Lortutako KTaren balioak positiboak dira, orduan %98eko konfiantza-mailaz muD (hau da, 
aiten batezbesteko altuera ken amen bateztebesteko altuera)
positiboa dela baieztatzen dugu eta, orduan, aiten batezbesteko altuera amen bateztebesteko altuera baino handiagoa dela ondorioztatzen dugu.

## Populazio binomial baten proportzioa

### Bizkaikoen proportzioaren KTa

In [135]:
# zenbat bizkaitar dauden zenbatzen dugu
Datuak.bizkaia <- subset(Datuak,JAIOTERRIA==1)
kopuru.bizkaia <- length(Datuak.bizkaia$ALTUERA)
#prop.bizkaia
guztiak <- length(Datuak$ALTUERA)
#guztiak
prop.test(kopuru.bizkaia,guztiak,conf.level=0.95)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 p = 0.5

Alternative Hypothesis:          True p is not equal to 0.5

Test Name:                       1-sample proportions test with continuity correction

Estimated Parameter(s):          p = 0.2280702

Data:                            kopuru.bizkaia out of guztiak, null probability 0.5

Test Statistic:                  X-squared = 49.49708

Test Statistic Parameter:        df = 1

P-value:                         1.986676e-12

95% Confidence Interval:         LCL = 0.1689802
                                 UCL = 0.2996563


#### Azalpena:
Bizkaikoen proportzioaren estimazioa: 0.2280702

Proportzioaren KTa: (0.1689802,0.2996563)



## Bi populazio binomialen proportzioen konparaketa

### Emakumeen eta gizonen bizkaitarren proportzioen diferentziaren KTa

In [136]:
# zenbat bizkaitar diren emakume eta gizon zenbatzen dugu
Datuak.bizkaia <- subset(Datuak,JAIOTERRIA==1)
emakume.bizkaia <- subset(Datuak.bizkaia,SEXUA=="emakumezkoa")
gizona.bizkaia <- subset(Datuak.bizkaia,SEXUA=="gizonezkoa")
n_emaku_biz <- length(emakume.bizkaia$ALTUERA)
n_gizon_biz <- length(gizona.bizkaia$ALTUERA)
# zenbat emakume eta gizon zenbatzen dugu
Datuak.emaku <- subset(Datuak,SEXUA=="emakumezkoa")
n_emaku <- length(Datuak.emaku$ALTUERA)
Datuak.gizon <- subset(Datuak,SEXUA=="gizonezkoa")
n_gizon <- length(Datuak.gizon$ALTUERA)
# testa egiten dugu
prop.test(c(n_emaku_biz,n_gizon_biz),c(n_emaku,n_gizon),conf.level=0.95)
#guztiak
#binom.test(kopuru.bizkaia,guztiak,conf.level=0.95)


Results of Hypothesis Test
--------------------------

Alternative Hypothesis:          two.sided

Test Name:                       2-sample test for equality of proportions with continuity correction

Estimated Parameter(s):          prop 1 = 0.2560976
                                 prop 2 = 0.2022472

Data:                            c(n_emaku_biz, n_gizon_biz) out of c(n_emaku, n_gizon)

Test Statistic:                  X-squared = 0.4303719

Test Statistic Parameter:        df = 1

P-value:                         0.5118065

95% Confidence Interval:         LCL = -0.08391627
                                 UCL =  0.19161701


#### Azalpena:
Emakumeen artean bizkaikoen proportzioaren estimazioa: 0.2560976

Gizonen artean bizkaikoen proportzioaren estimazioa: 0.2022472

Proportzioen diferentziaren KTa: (-0.08391627,0.19161701)

#### Ondorioa:
Zero KTaren barruan dagoenez, ezin dugu ondorioztatu bizkaitarren proportzioa ezberdina den gizonezkoen eta emakumeen artean
